cpp/ql/lib/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -827,82 +894,95 @@ private predicate unary_compares_eq(
   // ((test is `areEqual` => op == const + k2) and const == `k1`) =>
   // test is `areEqual` => op == k1 + k2
   inNonZeroCase = false and
-  exists(int k1, int k2, ConstantInstruction const |
+  exists(int k1, int k2, Instruction const |
```

Comment:
Why this change?

Reply:
Oh, that wasn't actually needed here, no. I've got an another upcoming PR to add value numbering to the `int_value` predicate (which currently takes a `ConstantInstruction`) in order to detect more constant cases, and that would need this particular change as part of it.

RIght now, this type relaxation doesn't change anything on this PR as the type of `const` is restricted by the call to `int_value`. I'm happy to undo this if you prefer, though

Reply:
It's fine to leave it as it is.